### PR TITLE
Feature/seamless immutable

### PR DESCRIPTION
**Pull request by joeyfigaro**
Implemented @chardskarth changes from https://github.com/supasate/connected-react-router/pull/72 so we can get this guy merged. 🍰 

Updated tests (rather naively) to get them to pass—updated the expectations of each failing test to match the expectations of the tests on master.

### CODE REVIEW

**Comment by supasate** [comment]
Thanks @joeyfigaro for working on this. Can you format the code by following this project's eslint rules?

**Comment by joeyfigaro** [comment]
@supasate definitely—should've checked before putting it up. I'll have those changes up shortly. 👍

**Comment by joeyfigaro** [comment]
@supasate I didn't find many problems outside of package.json and .eslintrc using tabs instead of spaces—did you happen to see anything else?

Steps I took:
- `./node_modules/.bin/eslint -c .eslintrc src/**/*`
- `./node_modules/.bin/eslint -c .eslintrc test/**/*`

Got zero results for both.

File: test/ConnectedRouter.test.js [hunk 1]
```diff
@@ -38,18 +39,18 @@ describe('ConnectedRouter', () => {
     props = {
       action: 'POP',
       location: {
-        pathname: '/path/to/somewhere',
+        pathname: '/path/to/somewhere'
```
Comment (supasate) [inline test/ConnectedRouter.test.js]: Leave the ending comma.

File: test/ConnectedRouter.test.js [hunk 2]
```diff
@@ -38,18 +39,18 @@ describe('ConnectedRouter', () => {
...
       },
-      history,
+      history
```
Comment (supasate) [inline test/ConnectedRouter.test.js]: Leave the ending comma.

File: test/ConnectedRouter.test.js [hunk 3]
```diff
@@ -38,18 +39,18 @@ describe('ConnectedRouter', () => {
...
     }
 
     // Mock store
     const mockStore = configureStore()
     store = mockStore({
       router: {
         action: 'POP',
-        location: props.history.location,
-      },
+        location: props.history.location
```
Comment (supasate) [inline test/ConnectedRouter.test.js]: Leave the ending comma.

File: test/ConnectedRouter.test.js [hunk 4]
```diff
@@ -38,18 +39,18 @@ describe('ConnectedRouter', () => {
...
+      }
```
Comment (supasate) [inline test/ConnectedRouter.test.js]: Leave the ending comma.

File: test/ConnectedRouter.test.js [hunk 5]
```diff
@@ -252,9 +292,9 @@ const storeShape = PropTypes.shape({
 
 ContextWrapper.propTypes = {
   store: storeShape.isRequired,
-  children: PropTypes.element.isRequired,
+  children: PropTypes.element.isRequired
```
Comment (supasate) [inline test/ConnectedRouter.test.js]: Leave the ending comma.

File: test/ConnectedRouter.test.js [hunk 6]
```diff
@@ -252,9 +292,9 @@ const storeShape = PropTypes.shape({
...
 }
 
 ContextWrapper.childContextTypes = {
-  store: storeShape.isRequired,
+  store: storeShape.isRequired
```
Comment (supasate) [inline test/ConnectedRouter.test.js]: Leave the ending comma.

File: test/reducer.test.js [hunk 7]
```diff
@@ -1,120 +1,172 @@
...
+import { combineReducers as combineReducersSeamlessImmutable } from 'redux-seamless-immutable'
 import Immutable from 'immutable'
 import { LOCATION_CHANGE, connectRouter } from '../src'
-import { connectRouter as connectRouterImmutable} from '../src/immutable'
+import { connectRouter as connectRouterImmutable } from '../src/immutable'
+import { connectRouter as connectRouterSeamlessImmutable } from '../src/seamless-immutable'
 
 describe('connectRouter', () => {
-  let mockHistory
+	let mockHistory
```
Comment (supasate) [inline test/reducer.test.js]: 2 spaces instead of a tab. Please fix all of the followings.

**Comment by joeyfigaro** [comment]
@supasate requested changes up!

**Comment by chardskarth** [comment]
Awesome. Just saw this today. ⭐ I wish I could've helped then.

**Pull request merged**